runtime/Java/src/org/antlr/v4/runtime/RuleContextWithAltNum.java

```diff
@@ -0,0 +1,20 @@
+package org.antlr.v4.runtime;
+
+/** A handy class for use with
+ *
+ *  options {contextSuperClass=org.antlr.v4.runtime.RuleContextWithAltNum;}
+ *
+ *  that provides a backing field / impl for the outer alternative number
+ *  matched for an internal parse tree node.
+ *
+ *  I'm only putting into Java runtime as I'm certain I'm the only one that
+ *  will really every use this.
+ */
+public class RuleContextWithAltNum extends ParserRuleContext {
```

Comment:
:thought_balloon: I think this needs to override `copyFrom`, or labeled outer alternatives won't work with it.